src/delfino/commands/test.py

```diff
@@ -28,45 +32,67 @@ def _run_tests(app_context: AppContext, name: str, maxfail: int, debug: bool) ->
 
     print_header(f"️Running {name} tests️", icon="🔎🐛")
     ensure_reports_dir(delfino)
-    run(
-        list(
-            filter(
-                None,
-                [
-                    "pytest",
-                    "--cov",
-                    delfino.sources_directory,
-                    "--cov-report",
-                    f"xml:{delfino.reports_directory / f'coverage-{name}.xml'}",
-                    "--cov-branch",
-                    "-vv",
-                    "--maxfail",
-                    str(maxfail),
-                    "-s" if debug else "",
-                    delfino.tests_directory / name,
-                ],
+
+    options: ArgsList = []
+    if debug:
+        options.append("--capture=no")
+        warnings.warn(
+            DeprecatedWarning(
+                "--debug for test-unit and test-integration command",
+                "0.19.0",
+                "1.0.0",
+                "Use --capture=no or -s instead.",
             )
-        ),
```

Comment:
I'm afraid this deprecation warning will not work very well for two reasons:
- There is no test using [`@fail_if_not_removed`](https://deprecation.readthedocs.io/en/latest/#deprecation.fail_if_not_removed) to force removal of the code at `1.x.y`
- The deprecation warning has no way of knowing it reached `1.x.y`+ because it is not given the current version. Therefore, it will likely not trigger [`@fail_if_not_removed`](https://deprecation.readthedocs.io/en/latest/#deprecation.fail_if_not_removed) even if it was here.

The purpose of this library is not only to notify users of upcoming removal but also to actually force the code removal later.

If you don't want to test this whole command, you could try abstracting the handling of the deprecated flag into a function and test only that function.

Reply:
I see, now I understand the purpose...

I decided not to deprecate the option with this PR.

I think the package is for function/method and won't support deprecating options. It may be possible to implement the warning by introducing a function to get the option value but I think it is too much for this PR.

Reply:
Removed deprecation with [e50f7f9](https://github.com/radeklat/delfino/pull/35/commits/e50f7f9def78e483b134a95c469a4adab8ec9ea3).